src/main/java/com/pkb/unit/Unit.java

```diff
@@ -107,6 +107,84 @@ public Unit(String id, Bus bus, long retryPeriod, TimeUnit retryTimeUnit) {
         handleReportState();
     }
 
+
+    /**
+     * @return true if this set did not already contain the specified element
+     */
+    public void addDependency(String dependency) {
+        mandatoryDependencies.put(dependency, Optional.empty());
+
+        // Publish our new list of dependencies
+        handleReportDependencies();
+
+        // Prompt the dependency to report it's state
+        unchecked(() -> bus.sink().accept(message(ReportStateRequest.class).withTarget(dependency)));
+    }
+
+    /**
+     * @return true if this set contained the specified element
+     */
+    public void removeDependency(String dependency) {
```

Comment:
The same as with addDependency so return some feedback.

Reply:
I'll change the doc here too

Reply:
I agree that we change it once the need comes up.